overlord/configstate/configcore/certs.go

```diff
@@ -32,6 +33,221 @@ import (
 	"github.com/snapcore/snapd/overlord/configstate/config"
 )
 
+type certificate struct {
+	Name     string
+	Path     string
+	RealPath string
+}
+
+// This function validates the name of the certificate. We only allow certificates
+// with the correct suffix ".crt" except for the ca-certificates.crt,
+// or if the name is in the blockedCerts lists.
+func isBlocked(cert certificate, blockedCerts []string) bool {
+	// Special case for ca-certificates.crt
+	if cert.Name == "ca-certificates.crt" {
```

Comment:
I would vote to have the caller add this explicitly to the `blockedCerts` list, otherwise I'm afraid that we are letting some "magic" leak into the code :-)

Or if we had an object to operate on, it could have an `AddBlockedCertificates()` method, and `isBlocked()` could then take no parameters.

But I miss the big picture of this, so it may be that what I'm asking does not make much sense.

Reply:
Yea I completely understand your confusion here. The thing is that we have yet to agree on _how_ it's supposed to be used. We are just imitating the functionality of update-ca-certificates here (but with support for aribtrary sources/destination), and the point of having this 'magic' here is because the ca-certificates.crt is a file we create in the output folder, hence there is never any need to copy it as it will be overriden in any case. (And such there is never any need for anyone to provide it as blocked)

Reply:
I see, thanks for the explanation!